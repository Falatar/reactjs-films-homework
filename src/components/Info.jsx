import React from 'react';
import style from './Info.scss';

function Info() {
  return (
    <div className={style.info}>
      <button type="button" id="watch">Watch now</button>
      <button type="button" id="check">View info</button>
    </div>
  );
}

export default Info;
