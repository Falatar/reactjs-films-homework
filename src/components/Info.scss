div.info {
    display: flex;
    flex-direction: row;
    border: 0;
    padding: 0;
    margin: 3vh 6vw 0 0;
}

.info {
    button {
        height: 40px;
        width: 120px;
        color: white;
        border-radius: 5px;
    }

    :first-child {
        border: 0;
        background-color: deepskyblue;

    }

    :last-child {
        border: 1px solid;
        background-color: transparent;
        margin-left: 0.6vw;
    }
}