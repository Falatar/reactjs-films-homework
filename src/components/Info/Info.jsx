import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './Info.scss';

class Info extends Component {
  constructor(props) {
    super(props);
    this.state = {
      textVisible: false,
    };
  }

  switchMode = () => {
    this.setState((state) => ({
      textVisible: !state.textVisible,
    }));
  }

  callModal = () => {
    const { callTrailer, setTrailer, id } = this.props;
    setTrailer(id);
    callTrailer();
  }

  render() {
    const { overview } = this.props;
    const { textVisible } = this.state;
    return (
      <div
        className={
          textVisible
            ? classNames(style.info, style.shadow)
            : classNames(style.info, style.sheer)
        }
      >
        <p
          className={
            textVisible
              ? classNames(style.text, style.visible)
              : classNames(style.text, style.unvisible)
          }
        >
          {overview}
        </p>
        <div className={style.buttons}>
          <button type="button" id="watch" className={style.watch} onClick={this.callModal}>Watch now</button>
          <button type="button" id="check" className={style.view} onClick={this.switchMode}>View info</button>
        </div>
      </div>
    );
  }
}

Info.defaultProps = {
  overview: '',
  id: 0,
  callTrailer: () => 'can\'t find callTrailer function',
  setTrailer: () => 'can\'t find setTrailer function',
};

Info.propTypes = {
  overview: PropTypes.string,
  id: PropTypes.number,
  callTrailer: PropTypes.func,
  setTrailer: PropTypes.func,
};

export default Info;
