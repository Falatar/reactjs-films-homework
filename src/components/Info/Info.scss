
.info {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    border: 0;
    padding: 0;
    margin: 3vh 6vw 0 0;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;

    .Watch, .View {
        height: 40px;
        width: 120px;
        color: white;
        border-radius: 5px;
        outline:none;
    }

    .Watch {
        border: 0;
        background-color: deepskyblue;

    }

    .View {
        border: 1px solid;
        background-color: transparent;
        margin-left: 0.6vw;
    }
}