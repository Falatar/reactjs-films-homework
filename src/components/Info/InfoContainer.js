import { connect } from 'react-redux';
import { updateTrailerPath, openModal } from '../../modules/singleFilmModule/singleFilmModuleActions';
import Info from './Info';


export const mapDispatchToProps = (dispatch) => ({
  callTrailer: () => dispatch(openModal()),
  setTrailer: (path) => dispatch(updateTrailerPath(path)),
});

export default connect(
  null,
  mapDispatchToProps,
)(Info);
