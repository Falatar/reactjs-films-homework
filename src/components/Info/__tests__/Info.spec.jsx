import React from 'react';
import renderer from 'react-test-renderer';
import Info from '../Info';

it('renders correctly', () => {
  const tree = renderer
    .create(<Info />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});