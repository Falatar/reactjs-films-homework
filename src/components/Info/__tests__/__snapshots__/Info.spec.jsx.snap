// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders correctly 1`] = `
<div
  className="info"
>
  <button
    className="Watch"
    id="watch"
    type="button"
  >
    Watch now
  </button>
  <button
    className="View"
    id="check"
    type="button"
  >
    View info
  </button>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="info"
>
  <button
    className="Watch"
    id="watch"
    type="button"
  >
    Watch now
  </button>
  <button
    className="View"
    id="check"
    type="button"
  >
    View info
  </button>
</div>
`;
